pkgs/development/tools/vulkan-validation-layers/default.nix

```diff
@@ -13,13 +13,30 @@
 , libXdmcp
 , libXrandr
 , spirv-headers
-, spirv-tools
 , vulkan-headers
 , wayland
 }:
 
 let
   robin-hood-hashing = callPackage ./robin-hood-hashing.nix {};
+
+  # Current VVL version requires a newer spirv-headers than the latest release tag.
+  # This should hopefully not be too common and the override should be removed after
+  # the next SPIRV headers release.
+  # FIXME: if this ever becomes common, figure out a way to pull revisions directly
+  # from upstream known-good.json
+  spirv-headers' = spirv-headers.overrideAttrs(_: {
```

Comment:
It's okay on how it is now. But next time a Nintendo emulator requires these amount of error-prone diffs, we should just override in it instead. The Vulkan SDK still looks more appropriate for a distro to supply.

Reply:
I agree that we should stick to SDK releases.

Reply:
It's not just a Nintendo emulator, it's entirely possible for those extensions to be used by other things as well, e.g. DXVK. It may be a better idea to build the validation layers and other development tools with the SDK versions of everything, but use the latest headers and loader by default.

Reply:
It's *possible* for stuff to use bleeding edge interfaces, yeah, but quite rare for people to front run the SDK in my experience. Stable nixpkgs will inevitably lag regardless; better that it lag at a well-defined, supported point.

Reply:
Filed upstream: https://github.com/KhronosGroup/SPIRV-Headers/issues/345